vaadin-testbench-core/src/main/kotlin/com/vaadin/testbench/unit/internal/ComponentUtils.kt

```diff
@@ -0,0 +1,393 @@
+//val HasComponents.isNotEmpty: Boolean get() = hasChildren
+
+/**
+ * True when the component has any children.
+ */
+val HasComponents.hasChildren: Boolean get() = (this as Component).children.findFirst().isPresent
+
+/**
+ * True when the component has no children.
+ *
+ * Deprecated: poorly named. A `form.isEmpty` may be ambiguous - it may refer to
+ * whether the form has any children, or whether the form value is empty, or something else.
+ */
+//@Deprecated("use !hasChildren")
+//val HasComponents.isEmpty: Boolean get() = !hasChildren
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * calls [HasStyle.addClassName] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.addClassNames2(classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    classNames.splitByWhitespaces().forEach { addClassName(it) }
+}
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * calls [addClassNames2] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.addClassNames2(vararg classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    classNames.forEach { addClassNames2(it) }
+}
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * calls [HasStyle.removeClassName] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.removeClassNames2(classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    classNames.splitByWhitespaces().forEach { removeClassName(it) }
+}
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * calls [removeClassNames2] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.removeClassNames2(vararg classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    classNames.forEach { removeClassNames2(it) }
+}
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * clears the class names and calls [addClassNames2] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.setClassNames2(classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    style.clear()
+    addClassNames2(classNames)
+}
+
+/**
+ * Splits [classNames] by whitespaces to obtain individual class names, then
+ * clears the class names and calls [addClassNames2] on each class name. Does nothing if the string
+ * is blank.
+ */
+fun HasStyle.setClassNames2(vararg classNames: String) {
+    // workaround for https://github.com/vaadin/flow/issues/11709
+    style.clear()
+    addClassNames2(*classNames)
+}
+
+/**
+ * A component placeholder, usually shown when there's no value selected.
+ * Not all components support a placeholder; those that don't will return null.
+ */
+var Component.placeholder: String?
```

Comment:
We should have these extensions that are not for all components so that the feature is findable for testing.

Reply:
Also for the components that have a placeholder the method is setPlaceholder so this wouldn't really give anything new on that part.